tests/app/shared/issue/duplicatedIssues/duplicated-issues.component.spec.ts

```diff
@@ -0,0 +1,147 @@
+        fixture.detectChanges();
+
+        debugElement = fixture.debugElement;
+        nativeElement = fixture.nativeElement;
+    });
+
+    it('should create a chip for each duplicated issue', () => {
+        // Check that the number of chips is the same as the number of duplicate issues
+        const matChipListLength: number = debugElement.queryAll(By.css('.mat-chip')).length;
+        const duplicateIssueslength: number = getDuplicateIssuesLength();
+        expect(matChipListLength).toEqual(duplicateIssueslength);
+
+        // Check that first chip contains information on first issue in the duplicate issues array
+        const matChipAnchor: HTMLElement = nativeElement.querySelector('a');
+        expect(matChipAnchor.innerText).toEqual(`#${firstDummyIssue.id}`);
+    });
+
+    it('should only allow cancellation of duplicate status if team/tutor response is editable', () => {
+        // Team/tutor response is not editable
+        permissionService.isTeamResponseEditable.and.callFake(() => false);
+        permissionService.isTutorResponseEditable.and.callFake(() => false);
+        fixture.detectChanges();
+        const cancelIconQuery = debugElement.query(By.css('.mat-icon'));
+        expect(cancelIconQuery).toBeNull();
+
+        // Team response is editable
+        permissionService.isTeamResponseEditable.and.callFake(() => true);
+        permissionService.isTutorResponseEditable.and.callFake(() => false);
+        fixture.detectChanges();
+        let cancelIcon: HTMLElement = debugElement.query(By.css('.mat-icon')).nativeElement;
+        expect(cancelIcon).toBeDefined();
+
+        // Tutor response is editable
+        permissionService.isTeamResponseEditable.and.callFake(() => false);
+        permissionService.isTutorResponseEditable.and.callFake(() => true);
+        fixture.detectChanges();
+        cancelIcon = debugElement.query(By.css('.mat-icon')).nativeElement;
+        expect(cancelIcon).toBeDefined();
+    });
+
+    it('should remove duplicate status of an duplicated issue once cancel icon is clicked', () => {
+        let updatedFirstDummyIssue: Issue;
+        issueService.updateIssueWithComment.and.callFake((duplicatedIssue: Issue) => {
+            updatedFirstDummyIssue = duplicatedIssue;
+            return of(duplicatedIssue);
+        });
+        spyOn(component, 'removeDuplicateStatus').and.callThrough();
+        permissionService.isTeamResponseEditable.and.callFake(() => true);
+        fixture.detectChanges();
+
+        expect(firstDummyIssue.duplicateOf).toEqual(component.issue.id);
+        cancelDuplicateStatus();
+        expect(component.removeDuplicateStatus).toHaveBeenCalledWith(firstDummyIssue);
+        expect(updatedFirstDummyIssue.duplicateOf).toBeNull();
+        expect(issueService.updateLocalStore).toHaveBeenCalledWith(updatedFirstDummyIssue);
+
+        // duplicate status of first dummy issue is removed
+        component.duplicatedIssues = of([secondDummyIssue]);
```

Comment:
Maybe this is not necessary, as the above checks will be sufficient. What do you think about this? ><

Reply:
This is actually because I would need to depend on the other functions of other services such as `updateLocalStore`, etc, and hence in my tests I am trying to remove the dependencies on these functions by using spy objects to ensure that these functions are called correctly on the correct `dummyIssue` without actually executing them. Therefore I need to kind of manually remove the first `dummyIssue` i.e. set the `duplicatedIssues` Observable to `secondDummyIssue`, (the object it is supposed to be after all the updates), and then check the UI updates on the page!

Let me know if you think there are any problems with this, because I am also not entirely sure if what I'm doing is right😅

Reply:
Ah I see what you did there. 

However, if the API to `updateLocalStore` was called, it would be good enough to test till there and not mock the update of `component.duplicateIssues` not sure what do yall think >< @anubh-v @ptvrajsk

Reply:
I agree with @seanlowjk on this. In this test, let's stop at verifying that `updateLocalStore` has been called.

In the first test, we've already tested that the component displays chips correctly, given an observable of duplicate issues.

Reply:
Alright, got it!!